### PR TITLE
Brave news android fixes (uplift to 1.36.x)

### PR DESCRIPTION
**Pull request by alexsafe**
Uplift of https://github.com/brave/brave-core/pull/12078
Resolves https://github.com/brave/brave-browser/issues/20773
Resolves https://github.com/brave/brave-browser/issues/20772
Resolves https://github.com/brave/brave-browser/issues/20410
Resolves https://github.com/brave/brave-browser/issues/20769
Resolves https://github.com/brave/brave-browser/issues/20771
Resolves https://github.com/brave/brave-browser/issues/20769
Resolves https://github.com/brave/brave-browser/issues/20768
Resolves https://github.com/brave/brave-browser/issues/20766

Uplift of https://github.com/brave/brave-core/pull/12155
Resolves https://github.com/brave/brave-browser/issues/20899
Resolves https://github.com/brave/brave-browser/issues/20900
Resolves https://github.com/brave/brave-browser/issues/20850
Resolves https://github.com/brave/brave-browser/issues/20901
Resolves https://github.com/brave/brave-browser/issues/19854

Uplift of https://github.com/brave/brave-core/pull/12255
Resolves https://github.com/brave/brave-browser/issues/21096

Pre-approval checklist: 
- [x] You have tested your change on Nightly. 
- [ ] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [x] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Comment by srirambv** [comment]
Verification passed on Oppo Reno 5 with Android 12 / Samsung Tab A with Android 10 running 1.37.55 x64 nightly build

Uplift of #12078
- Verification notes can be found here [#12078 (comment)](https://github.com/brave/brave-core/pull/12078#issuecomment-1042838336)

Uplift of #12155
- Verification notes can be found here [#12155 (comment)](https://github.com/brave/brave-core/pull/12155#issuecomment-1042883407)
- 
Uplift of #12255

- Verification notes can be found here [#12255 (comment)](https://github.com/brave/brave-core/pull/12255#issuecomment-1042893844)

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.36.x` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-browser/issues/21166.

**Pull request merged**
